doc/imageview_warmups.md

```diff
@@ -0,0 +1,329 @@
+If you're new to Reactive or functionally-reactive programming in
+general, it pays to spend some time exploring Reactive's programming
+paradigm.  This documents a set of experiments conducted while
+rewriting ImageView.jl using Reactive and Gtk; perhaps they will serve
+as useful demonstrations for others.
+
+## Core concept: deferred execution
+
+Actions that you take on Reactive nodes are `push!`ed to a queue and
+run at some later time:
+
+```jl
+julia> using Reactive
+
+julia> n = Node(0)
+Node{Int64}(0, nactions=0)
+
+julia> push!(n, 1)
+
+julia> value(n)
+0
+
+julia> Reactive.run_till_now()
+
+julia> value(n)
+1
+```
+
+You can cause these updates to be processed at regular intervals---if
+julia isn't busy doing something else---as follows:
+```jl
+eventloop = Timer(_->Reactive.run_till_now(), 0.001, 0.001)
```

Comment:
In the latest commits, I put this in [Reactive.__init__](https://github.com/JuliaLang/Reactive.jl/commit/2acb72efd6ce69f93e7b959eae9d69b19a4c77a8#diff-ac236c9f23e3c15683602f303d89d8cdR179).

The reason is, if one is using multiple packages that run `Reactive.run`, then they might end up stepping on each other's run calls. I also made it so that `push!(...)` takes an optional third argument which is a callback that will be called in case of an error... By default this callback prints the error to `STDERR`.

I was thinking the reason for exposing `Reactive.run` is to allow packages to watch for crashes, but this seems like a worse idea than having the onerror callback.